kong/db/strategies/off/init.lua

```diff
@@ -208,23 +208,14 @@ local function select_by_field(self, field, value, options)
     _, value = next(value)
   end
 
-  local ws_id
-
-  if field == "cache_key" then
-    -- align with cache_key insertion logic in _set_entity_for_txn
-    ws_id = get_default_workspace()
-
-  else
-    ws_id = workspace_id(schema, options)
-
-    local unique_across_ws = schema.fields[field].unique_across_ws
-    -- only accept global query by field if field is unique across workspaces
-    assert(not options or options.workspace ~= null or unique_across_ws)
-
-    if unique_across_ws then
-      ws_id = get_default_workspace()
-    end
-  end
+  local unique_across_ws = schema.fields[field].unique_across_ws
```

Comment:
`schema.fields["cache_key"].unique_across_ws` will raise runtime error for nil could index `unique_across_ws`

Reply:
done, thanks.